test-it/karma/javascriptinvoke/CloseReference.Test.js

```diff
@@ -0,0 +1,146 @@
+            expect(viPathParser('getObjectReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('noErrorInAndValidReferenceError.status')).toBeFalse();
+            expect(viPathParser('noErrorInAndValidReferenceError.code')).toBe(0);
+            expect(viPathParser('noErrorInAndValidReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('isNotAValidRefnum1')).toBeFalse();
+            expect(viPathParser('errorInAndValidReferenceError.status')).toBeTrue();
+            expect(viPathParser('errorInAndValidReferenceError.code')).toBe(100);
+            expect(viPathParser('errorInAndValidReferenceError.source')).toBe('error');
+            expect(viPathParser('isNotAValidRefnum2')).toBeFalse();
+            done();
+        });
+    });
+
+    it('makes a JavaScript Dynamic Reference invalid for further use', function (done) {
+        var viName = 'InvalidReferenceParameterVI';
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, jsCloseReferenceViaUrl);
+        var viPathParser = vireoRunner.createVIPathParser(vireo, viName);
+        vireoRunner.enqueueVI(vireo, viName);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+            expect(viPathParser('getObjectReferenceError.status')).toBeFalse();
+            expect(viPathParser('getObjectReferenceError.code')).toBe(0);
+            expect(viPathParser('getObjectReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('closeReferenceError.status')).toBeFalse();
+            expect(viPathParser('closeReferenceError.code')).toBe(0);
+            expect(viPathParser('closeReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('isNotAValidRefnum1')).toBeTrue();
+            expect(viPathParser('invalidReferenceError.status')).toBeTrue();
+            expect(viPathParser('invalidReferenceError.code')).toBe(1556);
+            expect(viPathParser('invalidReferenceError.source')).not.toBeEmptyString();
+            done();
+        });
+    });
+
+    it('can close an array of references of one dimension', function (done) {
+        var viName = 'ArrayOfReferencesVI';
+        var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, jsCloseReferenceViaUrl);
+        var viPathParser = vireoRunner.createVIPathParser(vireo, viName);
+        vireoRunner.enqueueVI(vireo, viName);
+
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+            expect(viPathParser('getObjectReferenceError.status')).toBeFalse();
+            expect(viPathParser('getObjectReferenceError.code')).toBe(0);
+            expect(viPathParser('getObjectReferenceError.source')).toBeEmptyString();
+            expect(viPathParser('closeReferenceError.status')).toBeFalse();
```

Comment:
Would be nice to have a wrapper to pass just 'closeReferenceError' and assert that there is no error. Asserts in the test method would be more succinct. Thoughts?

Reply:
The tests are not longer karma based. They are now vtr based and they don't have this problem.